### ISSUE TITLE
Auto-reload

### ISSUE THREAD

**Issue by afeld** [joeyespo/grip#19]
So happy I found this tool - have been using [Marked](http://markedapp.com/), but it doesn't give an accurate representation of GFM.  Thanks!

Anyway, as I was working on my README with the browser next to me, was thinking it would be awesome to have it watch the file for changes, then auto-refresh the browser.  Would you accept a PR if I worked on this?

**Comment by madflow** [comment]
I have made some success with the https://pypi.python.org/pypi/livereload library. I simply had to replace the grip.server.serve method:

```
def serve(path=None, host=None, port=None, gfm=False, context=None,
          username=None, password=None,
          render_offline=False, render_wide=False, render_inline=False):
    """
    Starts a server to render the specified file
    or directory containing a README.
    """
    app = create_app(path, gfm, context, username, password,
                     render_offline, render_wide, render_inline)

    # Set overridden config values
    if host is not None:
        app.config['HOST'] = host
    if port is not None:
        app.config['PORT'] = port

    # Run local server
    from livereload import Server
    server = Server(app.wsgi_app)
    server.serve(app.config['PORT'])
```

Maybe there are some pit falls  - I do not not know.

**Comment by bob-rossi** [comment]
I tried this patch and it works great.

**Comment by sayanee** [comment]
+1 :+1:

**Comment by joeyespo** [comment]
Implemented by @markbt in #135 :sparkles: :fireworks: :smile:

**Comment by joeyespo** [comment]
FYI, this feature was officially released n 4.0.0 :smile: 

Be sure to upgrade with `pip install --upgrade grip`.

**Comment by pdurbin** [comment]
Works great! Thank you @markbt!
